base/abstractdict.jl

```diff
@@ -527,6 +527,71 @@ function hash(a::AbstractDict, h::UInt)
     hash(hv, h)
 end
 
+"""
+    modify!(f, d::AbstractDict{K, V}, key) -> (old => new)
+
+Lookup and then update, insert or delete in one go without re-computing the hash.
+Return a pair of `old::Union{Some,Nothing}` and `new::::Union{Some,Nothing}`.
+
+`f` is a callable object that must take a single `old::Union{Some, Nothing}` argument and
+return a `new::Union{T, Some{T}, Nothing}` value, where `T` is a type [`convert`](@ref)-able
+to the value type `V`.  The value `Some(d[key])` is passed to `f` if `haskey(d, key)`;
+otherwise `nothing` is passed.  If `f` returns `nothing`, corresponding entry in the
+dictionary `d`  is removed.  If `f` returns non-`nothing` value `x`, `key => something(x)`
```

Comment:
```suggestion
otherwise `nothing` is passed.  If `f` returns `nothing`, the corresponding entry in the
dictionary `d`  is removed.  If `f` returns a non-`nothing` value `x`, `key => something(x)`
```